### PR TITLE
Add semihosting and SWO examples to mbed_override_console().

### PR DESCRIPTION
**Pull request by korjaa**
Related to https://github.com/ARMmbed/mbed-os/pull/10553

### CODE REVIEW

**Comment by korjaa** [comment]
@TeroJaasko semihosting example in docs.

File: docs/api/platform/FileHandle.md
Comment (kjbracey) [inline docs/api/platform/FileHandle.md]: The above example's text was talking about the case where a *target* is using SerialWireOutput.

Both the new examples are probably/definitely the case where the *developer/application* is deciding to redirect. (The semihosting would crash without debugger attached, so it's probably only a temporary thing by the developer, and the other one is talking about the target default not being suitable).

So those two should be overriding `mbed_override_console`, and/or maybe the text should be shuffled a bit. Otherwise you'd have a horrible collision in the case where the target had the `SerialWireOutput` override above, and you tried to override to semihosting.

In an out-of-the-box Mbed OS, the WEAK `target` defaults may or may not be already overridden by the target's support code - they're reserved for target code.

The WEAK non-`target` defaults are always ready to be overridden by application code on any target. Or some libraries may offer an option to do the override for you - eg "esp8266.provide_default". 

We could have `platform.localfilesystem.provide_default_console` or something to provide this example behaviour.

Comment (korjaa) [reply docs/api/platform/FileHandle.md]: Good good, I'll update this.

Comment (0xc0170) [reply docs/api/platform/FileHandle.md]: was this updated?

Comment (korjaa) [reply docs/api/platform/FileHandle.md]: Not yet.

Comment (korjaa) [reply docs/api/platform/FileHandle.md]: Now it makes sense to me, I was initially using the wrong `mbed_target_override_console()` in my `main.cpp` . I juggled the text around to better support this new understanding.

@kjbracey-arm any comments?

**Comment by korjaa** [comment]
~~Please don't merge this until I rebase the fixup commits. I'll rebase after the changes are approved.~~
Nvm, decided to rebase already.

**Review by kjbracey — APPROVED** [review]

**Comment by AnotherButler** [comment]
Is this relevant to both 5.12 and 5.13?

**Comment by korjaa** [comment]
`mbed_console_override()` was introduced in https://github.com/ARMmbed/mbed-os/commit/f79bfddcad7ec5891f6e13cb919a712cf42ebdc3 (tools-release-5.8.5) so if I'm not mistaken, this is relevant starting from 5.8.5.

**Pull request merged**
